### ISSUE TITLE
Make nested relation loading work with same top level relation name

### ISSUE THREAD

**Issue by kiaking** [vuex-orm/vuex-orm#104]
Currently, if you have multiple nested relationship loading with `with` method, it doesn't work and you have to pass closure and nest the query.

```js
// This kind of call wouldn't wotk. Same `posts` on each `with` method.
store.getters['entities/workspaces/query']()
  .with('posts.comments')
  .with('posts.likes')
  .get()

// You have to do this.
store.getters['entities/workspaces/query']()
  .with('posts', (query) => {
    query.with('comments').with('likes')
  })
  .get()
```

I saw many people having trouble with not able to write prior syntax. S, I think we should make it able to do that.

**Comment by timoschwarzer** [comment]
Because it was easier to implement, I added a new syntax to the `with` method which can fetch multiple sub relations of a relation, e.g. `with('posts.comments|likes'`.
See: https://github.com/vuex-orm/vuex-orm/pull/107

**Comment by kiaking** [comment]
@timoschwarzer Thank you for your hard work! It's awesome.

I really like `with('posts.comments|likes')` syntax and I think this is the way to define it, but I still think we should support the separated `with` define also since people would likely to write like that ; ).

```js
.with('posts.comments')
.with('posts.likes')
```

**Comment by timoschwarzer** [comment]
@kiaking I will make a PR for that in an hour or two.

**Comment by kiaking** [comment]
@timoschwarzer NIICE! Thank you so much! Looking forward to it 👍

**Comment by paparent** [comment]
Hi,

That's a great addition ! Good job @timoschwarzer  ! 

I don't know if it's easy to implement (or maybe it's done) : 

It would be nice if we could load nested relations, and the last is a closure. Like this: 

`.with('brands.products', (query) => {...})`

**Comment by kiaking** [comment]
Sadly it doesn't work right now. Maybe we could make it to work that way though. It might be a nice addition.

**Comment by timoschwarzer** [comment]
I think we should not allow this type of relationship loading. We should rather detect when this syntax is used and throw a warning in the dev build of vuex-orm.

**Comment by kiaking** [comment]
@timoschwarzer Well yea maybe. I wonder what Laravel does for the eager load constraints on nested with clause by dot syntax. But I agree that it might be confusing. It is not very clear if that constraint is applied to `brands` or `products`.

**Comment by timoschwarzer** [comment]
@kiaking Oh, my message was confusing. It wasn't a reply to @paparent but to the syntax of the OP message.

**Comment by kiaking** [comment]
Oh no my bad. Maybe yeah. I would like to know how Laravel handles this though. It's not documented so I need to the check.

One valid use case that I can think of using this syntax is when you need to construct query builder function that dynamically builds up the query chain. In such case, maybe it's useful to have this syntax.

```js
const query = this.$store.getters['entities/users/query']().with('posts.comment')

if (getLikesToo) {
  query.with('posts.likes')
}

return query.get()
```

Obviously you could workaround above case in other ways, but it might be nice to have.

**Comment by kiaking** [comment]
OK now this feature will be available in next release! 🚀

**Comment by kiaking** [comment]
This feature is now available with Vuex ORM 0.25.5! 🎉 
https://github.com/vuex-orm/vuex-orm/releases/tag/v0.25.5
